overlord/configstate/configcore/ctrlaltdel_test.go

```diff
@@ -0,0 +1,165 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/overlord/configstate/configcore"
+	"github.com/snapcore/snapd/snap"
+)
+
+type unitStates int
+
+// The possible unit states we should test for to make sure
+// appropriate error messages are displayed
+const (
+	unitStateNone unitStates = iota
+	unitStateMulti
+	unitStateUninstalled
+	unitStateDisabled
+	unitStateEnabled
+	// Ubuntu Core <= 18 has an earlier version of systemd and the
+	// UnitFileState for a masked unit is returned as 'bad'.
+	// LoadState (unused by us) returns 'masked'.
+	unitStateMaskedv1
+	// Ubuntu Core > 18 has a later version of systemd and the
+	// UnitFileState for a masked unit is returned as 'masked'.
+	unitStateMaskedv2
+)
+
+type ctrlaltdelSuite struct {
+	configcoreSuite
+	unit unitStates
+}
+
+var _ = Suite(&ctrlaltdelSuite{})
+
+func (s *ctrlaltdelSuite) SetUpTest(c *C) {
+	s.configcoreSuite.SetUpTest(c)
+	s.systemctlOutput = func(args ...string) []byte {
+		var output []byte
+		if args[0] == "show" {
+			switch s.unit {
+			case unitStateMulti:
+				output = []byte("Id=ctrl-alt-del.target\nActiveState=inactive\nUnitFileState=enabled\nNames=ctrl-alt-del.target\n" +
+					"\n" +
+					fmt.Sprintf("Id=%s\nActiveState=inactive\nUnitFileState=enabled\nNames=%[1]s\n", args[2]))
```

Comment:
super nitpick: at first sight it's not obvious what `arg[2]` is; I'd suggest adding something like this to the line just before `switch`:
```
unitName := args[len(args)-1]
```
(explicitly getting the last argument in case we invoke systemd with more options later)

Reply:
I have added some comments here which I hope will be good enough. My opinion is that the context of the test is too specific (will only ever request the status for one unit named ctrl-alt-del.target), to really benefit from making the input more generic. Please let me know if you still feel the comments are not enough.